Author: Samuel Tardieu <sam@rfc1149.net>
Date:   Tue Oct 16 12:31:58 2007 +0200

    MAC_ARG is not defined by the kernel headers anymore

diff --git a/ipw3945.c b/ipw3945.c
index eec20b7..cc36b9c 100644
--- a/ipw3945.c
+++ b/ipw3945.c
@@ -102,6 +102,12 @@
 #define VQ
 #endif
 
+/* stolen from net/ieee80211.h */
+#ifndef MAC_ARG
+#define MAC_ARG(x) ((u8*)(x))[0], ((u8*)(x))[1], ((u8*)(x))[2], \
+		   ((u8*)(x))[3], ((u8*)(x))[4], ((u8*)(x))[5]
+#endif
+
 #define IPW3945_VERSION "1.2.2" VD VM VP VR VQ
 
 #define DRV_DESCRIPTION	"Intel(R) PRO/Wireless 3945 Network Connection driver for Linux"



Author: Samuel Tardieu <sam@rfc1149.net>
Date:   Tue Oct 16 12:37:54 2007 +0200

    Nuke SET_MODULE_OWNER() use

diff --git a/ipw3945.c b/ipw3945.c
index cc36b9c..0369876 100644
--- a/ipw3945.c
+++ b/ipw3945.c
@@ -16597,7 +16597,6 @@ static int ipw_pci_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 		goto out_release_ucode;
 	}
 
-	SET_MODULE_OWNER(net_dev);
 	SET_NETDEV_DEV(net_dev, &pdev->dev);
 	mutex_lock(&priv->mutex);
 	priv->ieee->hard_start_xmit = ipw_net_hard_start_xmit;

